src/lavinmq/launcher.cr

```diff
@@ -27,28 +41,36 @@ module LavinMQ
       end
       Dir.mkdir_p @config.data_dir
       if @config.data_dir_lock?
-        @data_dir_lock = DataDirLock.new(@config.data_dir).tap &.acquire
+        @data_dir_lock = DataDirLock.new(@config.data_dir)
       end
-      @replicator = replicator = etcd ? Clustering::Server.new(@config, etcd) : Clustering::NoopServer.new
-      @amqp_server = LavinMQ::Server.new(@config, replicator)
-      @http_server = LavinMQ::HTTP::Server.new(@amqp_server)
+
+      if @config.clustering?
+        etcd = Etcd.new(@config.clustering_etcd_endpoints)
+        @runner = controller = Clustering::Controller.new(@config, etcd)
+        @replicator = Clustering::Server.new(@config, etcd, controller.id)
```

Comment:
I think we should make `Clustering::Server` accept a `Clustering::Controller` instead of etcd and id, and then add methods to `Clustering::Controller` to handle cluster secret and isr. That would isolate all etcd code to `Clustering::Controller`.

Reply:
ok, but you would like to do that in another PR or?

Reply:
Yes